trieste/models/interfaces.py

```diff
@@ -666,3 +666,20 @@ def update_trajectory(self, trajectory: TrajectoryFunction) -> TrajectoryFunctio
         :return: The new trajectory function updated for a new model
         """
         return self.get_trajectory()
+
+
+@runtime_checkable
+class SupportsGetInducingVariables(ProbabilisticModel, Protocol):
+    """A probabilistic model uses and has access to an inducing point approximation."""
+
+    @abstractmethod
+    def get_inducing_variables(self) -> Tuple[TensorType, TensorType, TensorType, bool]:
```

Comment:
We're not super-consistent, but we seem to prefer using tuple rather than Tuple in type annotations (it requires `from __future__ import annotations` but that's already included here).

```suggestion
    def get_inducing_variables(self) -> tuple[TensorType, TensorType, TensorType, bool]:
```

Arguably it would be clearer to use a named tuple or dataclass here rather than a 4-tuple, but probably not worth the extra code.